docs/servicedns-with-externaldns.md

```diff
@@ -0,0 +1,250 @@
+# Multi-Cluster Service DNS with ExternalDNS Guide
+
+Multi-Cluster Service DNS (MCSDNS) provides the ability to programmatically manage DNS resource records of Service
+objects. MCSDNS is not meant to replace in-cluster DNS providers such as [CoreDNS](https://coredns.io/). Instead, MCSDNS
+integrates with [ExternalDNS](https://github.com/kubernetes-incubator/external-dns) through
+[CRD Source](https://github.com/kubernetes-incubator/external-dns/blob/master/docs/contributing/crd-source.md) to manage
+DNS resource records of Service objects in supported DNS providers.
+
+## Concepts
+
+<p align="center"><img src="./images/servicedns-with-externaldns.png" width="711"></p>
+
+The above diagram illustrates MCSDNS. A typical MCSDNS workflow consists of:
+
+1. Creating `FederatedDeployment` and `FederatedService` objects. The Federation sync
+controller propagates the corresponding objects to target clusters.
+2. Creating a `MultiClusterServiceDNSRecord` object that identifies the intended domain name and other optional resource
```

Comment:
s/MultiClusterServiceDNSRecord/ServiceDNSRecord

The rename PR https://github.com/kubernetes-sigs/federation-v2/pull/331 has been merged, so here we may need to update the concept here as well.